src/pages/Video.js

```diff
@@ -0,0 +1,89 @@
+import { useParams } from "react-router-dom";
+import { ReactComponent as ThumbsUpIcon } from "../assets/thumbs-up.svg";
+import { ReactComponent as ThumbsDownIcon } from "../assets/thumbs-down.svg";
+import Navbar from "../components/Navbar";
+import { useEffect, useState } from "react";
+import Spinner from "../components/Spinner";
+import { apiBaseURL } from "../constants";
+import VideoCard from "../components/VideoCard";
+
+export default function Video() {
+  const { id } = useParams();
+
+  const [video, setVideo] = useState(null);
+  const [videoLoading, setLoading] = useState(true);
+  const [recommendedVideos, setRecommendedVideos] = useState([]);
+  const [recommendedVideosLoading, setRecommendedVideosLoading] =
+    useState(true);
+
+  useEffect(() => {
+    fetch(`${apiBaseURL}/${id}`)
+      .then((r) => r.json())
+      .then((video) => setVideo(video))
+      .finally(() => setLoading(false));
+  }, [id]);
+
+  useEffect(() => {
+    if (video) {
+      fetch(`${apiBaseURL}`)
+        .then((r) => r.json())
+        .then((videos) => videos.filter((v) => v.id !== video.id))
+        .then(setRecommendedVideos)
+        .finally(() => setRecommendedVideosLoading(false));
+    }
+  }, [video, setRecommendedVideos]);
+
+  return videoLoading || !video ? (
+    <div className="w-screen h-screen bg-primary flex items-center justify-center">
+      <Spinner />
+    </div>
+  ) : (
```

Comment:
I highly suggest ditching the ternary here and instead breaking it into an if block:

```javascript
// If the video is null or loading, show the loading screen
if (videoLoading || !video) {
  return (
    <div className="w-screen h-screen bg-primary flex items-center justify-center">
      <Spinner />
    </div>
  )
}

// Otherwise, return the video container
return (
  <div className="w-screen min-h-screen bg-primary text-text-primary overflow-x-hidden">
  // ...
)
```

Reply:
This also helps people see that we're just returning html from this function bc people tend to forget that

Reply:
Agreed.